membership-attribute-service/app/controllers/AccountController.scala

```diff
@@ -129,10 +126,11 @@ class AccountController(
             subscription.termEndDate,
           )
           result = cancellationEffectiveDate.getOrElse("now").toString
+          catalog <- EitherT.rightT(services.futureCatalog)
           _ <- sendSubscriptionCancelledEmail(
             request.user.primaryEmailAddress,
             contact,
-            subscription.plan,
+            subscription.plan(catalog).productType(catalog),
```

Comment:
`subscription` was a high level but is now a low level one.
Where fields no longer exist, I have created a similarly named method, which takes the catalog and generates the old object on the fly.

regarding adding `.productType(catalog)` in this case we were passing in the high level rate plan object, but only using the productType, so I've narrowed the method parameter